### PR TITLE
Enabled CUDA support option

### PR DESCRIPTION
**Pull request by alexanderwatanabe**
Previous commit/push was to master branch. Read docs that I should avoid working out of master, sorry.
Also, set default of cudaSupport to false, (in master I had set it to true) as it seems like a bad idea to assume nvidia card for other users.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/dlib/default.nix
Comment (Ma27) [inline pkgs/development/libraries/dlib/default.nix:6]: Hmm, I'm wondering if we want to add a `dlib-nvidia` package to `pkgs/top-level/all-packages.nix` which is basically `dlib.override { cudaSupport = true; }`, otherwise, users have to build `dlib` with CUDA locally.

**Review by Ma27 — CHANGES_REQUESTED** [review]
@alexanderwatanabe thanks a lot for caring about the CUDA integration here! However, I'm afraid you misunderstood the contribution process for `nixpkgs`. Let me try to clarify a few things:

* it's fine to file a PR **against** master. This is where all the development happens and the `nixos-unstable`-[channel](https://nixos.wiki/wiki/Nix_channels) is created from our `master`. Every "new" change should somehow land on master before appearing on stable branches.
* you should avoid though to push the changes to your fork's master, instead it's better to create a feature-branch from *your* master-branch for the following reasons:
  * if the branch name is a bit more descriptive it makes the merge-commit more descriptive as well
  * a lot of people have further customizations on their master or somehow rely on their fork's master which can be a problem if a maintainer pushes minor fixes to a contributor's master before merging a PR.
* In case you want to [backport](https://gist.github.com/grahamc/c60578c6e6928043d29a427361634df6#what-to-backport) your changes, please **don't** file your PR against `nixos-20.03`, but against `release-20.03` instead. The `nixos-*` branches are created automatically for our channels. You can read more about this on https://howoldis.herokuapp.com/

Does this help you? If not, feel free to ping me either here or in the IRC :)

File: pkgs/development/libraries/dlib/default.nix
Comment (alexanderwatanabe) [reply pkgs/development/libraries/dlib/default.nix:6]: Thank you, you cleared things up for me. Sorry for not understanding the process before using your time.
In case it was not clear, I am new to both OSS contributing as well as nix/nixpkgs/nixos, so I appreciate your feedback and direction.

Can we reject this PR and refile under the appropriate branch, or is there some other mechanism to move it to the appropriate place?

One small touch of confusion: it's not 100% clear to me under what scenarios I should PR against release-20.03 vs. master (given your wording, is it safe to assume I can just file against master until I know of a reason to use release-20.03?)

Comment (Ma27) [reply pkgs/development/libraries/dlib/default.nix:6]: > Thank you, you cleared things up for me. Sorry for not understanding the process before using your time.
In case it was not clear, I am new to both OSS contributing as well as nix/nixpkgs/nixos, so I appreciate your feedback and direction.

No worries, everyone of us was a beginner at some point :) 

> Can we reject this PR and refile under the appropriate branch, or is there some other mechanism to move it to the appropriate place?

I'm afraid yes: we could change the branch to merge onto, but not your base branch (i.e. the branch from your fork with your CUDA changes).

> One small touch of confusion: it's not 100% clear to me under what scenarios I should PR against release-20.03 vs. master (given your wording, is it safe to assume I can just file against master until I know of a reason to use release-20.03?)

Basic criteria is listed here: https://gist.github.com/grahamc/c60578c6e6928043d29a427361634df6#what-to-backport (you may want to take a look at this as well: https://nixos.wiki/wiki/Maintainers:Maintainers)

But those are no hard rules, so in case you have a good reason, feel free to open a PR anyway to discuss :)

Comment (alexanderwatanabe) [reply pkgs/development/libraries/dlib/default.nix:6]: Thank you very much again. I appreciate your time greatly.

**Pull request closed without merging**
